fabric-convention-tags-v2/src/main/java/net/fabricmc/fabric/api/tag/convention/v2/FabricTagKeyExtension.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.tag.convention.v2;
+
+import net.minecraft.registry.tag.TagKey;
+import net.minecraft.text.Text;
+import net.minecraft.util.Identifier;
+
+/**
+ * General-purpose Fabric-provided extensions for {@link TagKey} subclasses.
+ *
+ * <p>These extensions were designed primarily for giving extra utility methods for TagKeys usages.
+ * Getting a TagKey's translation key for example.
+ *
+ * <p>Note: This interface is automatically implemented on all {@link TagKey} instances via Mixin and interface injection.
+ */
+public interface FabricTagKeyExtension {
```

Comment:
Should be called `FabricTagKey` like `FabricItem`, `FabricItemStack`, `FabricEntityTypeBuilder`, etc.